script.js

```diff
@@ -54,83 +65,276 @@ function createEpisodeDescription(tvEpisodeDiv, episode) {
   tvEpisodeDiv.appendChild(descDiv);
 }
 
-
-function makePageForEpisodes(episodeList) {
+function makePageForEpisodes(allEpisodes) {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  const episodeCounter = document.getElementById("showsAndEpisodesCounter");
+  episodeCounter.textContent = `Got ${allEpisodes.length} episode(s)`;
 
-  episodeList.forEach(episode => {
-    
+  allEpisodes.forEach((episode) => {
     const tvEpisodeDiv = createTVEpisodeDiv();
     createEpisodeTitle(tvEpisodeDiv, episode);
-    createEpisodeImage(tvEpisodeDiv, episode);
-    createEpisodeDescription(tvEpisodeDiv, episode);
+
+    //check if medium property is present
+    if (episode?.image?.medium != undefined) {
+      createEpisodeImage(tvEpisodeDiv, episode);
+    }
+
+    //check if summary property is present
+    if (episode.summary != undefined) {
+      createEpisodeDescription(tvEpisodeDiv, episode);
+    }
 
     rootElem.appendChild(tvEpisodeDiv);
   });
 }
 
+// --------------------------------TV SHOWS -----------------------------------
+
+//create show div el
```

Comment:
sometimes it really needed to break work into pieces to understand clearer, but if I would say I wouldn't give function for every creation of the element, because they all represent one card and but everyone looks different just wanna give my viewpoint, thanks a lot!